### PR TITLE
sql connection pool

### PR DESCRIPTION
**Pull request by Bond-009**
Should improve #9634

### CODE REVIEW

File: Emby.Server.Implementations/Data/BaseSqliteRepository.cs [hunk 1]
```diff
@@ -101,63 +110,115 @@ protected BaseSqliteRepository(ILogger<BaseSqliteRepository> logger)
...
-            WriteConnection.Execute("VACUUM");
+            writeConnection.Execute("VACUUM");
 
-            return new ManagedConnection(WriteConnection, WriteLock);
+            return writeConnection;
+        }
+
+        protected SQLiteDatabaseConnection CreateReadConnection()
+        {
+            var writeConnection = SQLite3.Open(
```
Comment (AmbulantRex) [inline Emby.Server.Implementations/Data/BaseSqliteRepository.cs:184]:
```suggestion
            var readConnection = SQLite3.Open(
```

Just a minor inconsistency with the naming.

File: Emby.Server.Implementations/Data/ConnectionPool.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+using SQLitePCL.pretty;
+
+namespace Emby.Server.Implementations.Data;
+
+/// <summary>
+/// A pool of SQLite Database connections.
+/// </summary>
+public sealed class ConnectionPool : IDisposable
+{
+    private readonly BlockingCollection<SQLiteDatabaseConnection> _connections = new();
```
Comment (crobibero) [inline Emby.Server.Implementations/Data/ConnectionPool.cs:12]: I don't think it _really_ matters but `BlockingCollection` is disposable

**Review by crobibero — APPROVED** [review]

File: Emby.Server.Implementations/Data/ConnectionPool.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        if (_disposed)
+        {
+            ThrowObjectDisposedException();
+        }
+
+        return new ManagedConnection(_connections.Take(), this);
+
+        void ThrowObjectDisposedException()
+        {
+            throw new ObjectDisposedException(GetType().Name);
```
Comment (cvium) [inline Emby.Server.Implementations/Data/ConnectionPool.cs:43]: Wouldn't nameof be better? Compile-time vs runtime?

**Review by cvium — APPROVED** [review]

File: Emby.Server.Implementations/Data/ConnectionPool.cs (hunk 3 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Data/ConnectionPool.cs:43]: Since this class is sealed, yes that would be better

**Pull request merged**